src/main/java/graphql/util/TreeTransformer.java

```diff
@@ -51,4 +55,38 @@ public TraversalControl leave(TraverserContext<T> context) {
         NodeMultiZipper<T> multiZipper = NodeMultiZipper.newNodeMultiZipperTrusted(root, zippers, nodeAdapter);
         return multiZipper.toRootNode();
     }
+
+    public T transformParallel(T root, TraverserVisitor<T> traverserVisitor) {
+        return transformParallel(root, traverserVisitor, Collections.emptyMap());
+
+    }
+
+    public T transformParallel(T root, TraverserVisitor<T> traverserVisitor, Map<Class<?>, Object> rootVars) {
+        assertNotNull(root);
+
+
+        TraverserVisitor<T> nodeTraverserVisitor = new TraverserVisitor<T>() {
+
+            @Override
+            public TraversalControl enter(TraverserContext<T> context) {
+                NodeZipper<T> nodeZipper = new NodeZipper<>(context.thisNode(), context.getBreadcrumbs(), nodeAdapter);
+                context.setVar(NodeZipper.class, nodeZipper);
+                context.setVar(NodeAdapter.class, nodeAdapter);
+                return traverserVisitor.enter(context);
+            }
+
+            @Override
+            public TraversalControl leave(TraverserContext<T> context) {
+                return Assert.assertShouldNeverHappen("not supposed to be called for parallel traversing");
+            }
+        };
+
+        Queue<NodeZipper<T>> zippers = new ConcurrentLinkedQueue<>();
+        ParallelTraverser<T> traverser = ParallelTraverser.parallelTraverserWithNamedChildren(nodeAdapter::getNamedChildren, zippers);
```

Comment:
So now  the default choice of the `ForkJoin` pool is still the platform common pool.

I think it needs to be "varially" be made available to the outer most limit.  Because while the platform default is sensible in most cases - its not for graphql-java to choose.